BLUESPAWN-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -217,4 +217,85 @@ namespace EventLogs {
 		return detect;
 	}
 
+	bool IsChannelOpen(const std::wstring& channel) {
+		EVT_HANDLE hChannel = NULL;
+		DWORD status = ERROR_SUCCESS;
+		PEVT_VARIANT pProperty = NULL;  
+		PEVT_VARIANT pTemp = NULL;
+		DWORD dwBufferSize = 0;
+		DWORD dwBufferUsed = 0;
+
+		// Open the channel config
+		hChannel = EvtOpenChannelConfig(NULL, channel.c_str(), 0);
+		if (NULL == hChannel)
+		{
+			LOG_ERROR(L"EventLogs::IsChannelOpen: EvtOpenChannelConfig failed with " + std::to_wstring(GetLastError()) + L" for channel " + channel);
+			return false;
+		}
+
+		// Attempt to get the channel property
+		if (!EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed))
+		{
+			status = GetLastError();
+			if (ERROR_INSUFFICIENT_BUFFER == status) {
+				dwBufferSize = dwBufferUsed;
+				pTemp = (PEVT_VARIANT)realloc(pProperty, dwBufferSize);
+
+				if (pTemp) {
+					pProperty = pTemp;
+					pTemp = NULL;
+					EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed);
+				}
+				else {
+					if (pProperty)
+						free(pProperty);
+
+					LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+					return false;
+				}
+			}
+
+			if (ERROR_SUCCESS != (status = GetLastError())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5037686)